lib/backend/watchersyncer/watchercache.go

```diff
@@ -432,13 +410,12 @@ func (wc *watcherCache) markAsValid(resourceKey string) {
 	}
 }
 
-// onError signals to the syncer that this watcherCache is not in-sync if the number of consecutive errors
-// exceeds the error threshold.  See finishResync() for how the watcherCache goes back to in-sync.
-func (wc *watcherCache) onError() {
-	wc.errors++
-	if wc.hasSynced && wc.errors > wc.errorThreshold {
-		wc.logger.WithFields(logrus.Fields{"errors": wc.errors, "threshold": wc.errorThreshold}).Debugf("Exceeded error threshold")
-		wc.hasSynced = false
+// If a syncer isn't in-sync state, the onWaitForDatastore method, sends WaitForDatastore event.
+// See finishResync() for how the watcherCache goes back to in-sync.
+func (wc *watcherCache) onWaitForDatastore() {
+	wc.logger.Debug("Send WaitforDatastore event if sync isn't in-sync")
+	if !wc.hasSynced {
```

Comment:
Should br `if wc.hasSynced`  -  i.e. we only indicate that we are going out-of-sync if we are already synced

Reply:
Updated PR to reflect as per suggestions.